unity/src/main/java/io/delta/unity/UCCatalogManagedClient.java

```diff
@@ -59,56 +57,92 @@ public UCCatalogManagedClient(UCClient ucClient) {
   // TODO: [delta-io/delta#4817] loadTable API that takes in a UC TableInfo object
 
   /**
-   * Loads a Kernel {@link ResolvedTable} at a specific version.
+   * Loads a Kernel {@link ResolvedTable}. If no version is specified, the latest version of the
+   * table is loaded.
    *
    * @param engine The Delta Kernel {@link Engine} to use for loading the table.
    * @param ucTableId The Unity Catalog table ID, which is a unique identifier for the table in UC.
    * @param tablePath The path to the Delta table in the underlying storage system.
-   * @param version The version of the table to load.
+   * @param versionOpt The optional version of the table to load.
    */
-  public ResolvedTable loadTable(Engine engine, String ucTableId, String tablePath, long version) {
+  public ResolvedTable loadTable(
+      Engine engine, String ucTableId, String tablePath, Optional<Long> versionOpt) {
     Objects.requireNonNull(engine, "engine is null");
     Objects.requireNonNull(ucTableId, "ucTableId is null");
     Objects.requireNonNull(tablePath, "tablePath is null");
-    checkArgument(version >= 0, "version must be non-negative");
+    Objects.requireNonNull(versionOpt, "versionOpt is null");
+    versionOpt.ifPresent(version -> checkArgument(version >= 0, "version must be non-negative"));
 
-    logger.info("[{}] Resolving table at version {}", ucTableId, version);
-    final GetCommitsResponse response = getRatifiedCommitsFromUC(ucTableId, tablePath, version);
-    validateLoadTableVersionExists(ucTableId, version, response.getLatestTableVersion());
+    logger.info(
+        "[{}] Loading ResolvedTable at version {}", ucTableId, getVersionString(versionOpt));
+    final GetCommitsResponse response = getRatifiedCommitsFromUC(ucTableId, tablePath, versionOpt);
+    final long ucTableVersion = getTrueUCTableVersion(ucTableId, response.getLatestTableVersion());
+    versionOpt.ifPresent(
+        version -> validateLoadTableVersionExists(ucTableId, version, ucTableVersion));
```

Comment:
is this expensive, should it be included in the timing?

Reply:
not expensive, just an `if` statement 👍